dll/win32/browseui/settings.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * PROJECT:     ReactOS BrowseUI
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Stores settings for the file explorer UI.
+ * COPYRIGHT:   Copyright 2023 Carl Bialorucki <cbialo2@outlook.com>
+ */
+
+#include "precomp.h"
+
+BrowseUISettings gSettings;
```

Comment:
This needs to be initialized here, or the struct needs a ctor.
Otherwise you are using uninitialized memory.

Reply:
Added a constructor

Reply:
Question: Should this be a global? Or a member of CShellBrowser?

Reply:
I think it should be global because in addition to `CShellBrowser` I already also need it in `CInternetToolbar`.  I would expect many other settings to be littered throughout browseui. The setting controlling if control panel is shown under 'My Computer' will probably also have to be done outside of `CShellBrowser`.

Reply:
Since you're making it a global, does this mean that the constructor is called _**very early on**_ (by the CRT) when the dll is being loaded? Should it be so? Or should we wait until the DllMain() gets called (or even after gModule.Init() is called, see https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/browseui/browseui.cpp;h=8634429bfc95b174a58fd397ddf8ec8f0f70a609;hb=HEAD#l170 )

Reply:
If it's a simple ctor (just setting values) it is fine to just have it as global, when it is doing complex stuff that _might_ depend on initialization in the rest of the DLL, it should not be a global because it runs before DllMain gets a chance to initialize stuff indeed.

Reply:
When my comment was made, the constructor called the Load() function, which were doing reads from the registry...

Reply:
I decided to use the default constructor to initialize the settings struct and to call `Load()` when shellbrowser is initialized.

Reply:
Which default constructor?
You did not define one, you just call it (which should be called implicitly already when constructing the object), now you also invoke the copy constructor or assignment operator, whilst still not initializing your member variables.

Reply:
If you have 100 member functions that have to act on all the members, for sure that adding yet a new member may require modifying all those functions :thinking:

Reply:
> I added a constructor and initialized each member variable in it. Is there a better way to do this? Now to add a new setting you have to update four places (the header, the ctor, the load function, and the save function).

If this module is compiled as c++11 code you can use in-class initializers, which means you can drop the entire constructor.

Then it would look like:
```
struct BrowseUISettings
{
    BOOL fLocked = FALSE;
    BOOL fShowGoButton = FALSE;
    BOOL fStatusBarVisible = FALSE;

    BOOL Load();
    BOOL Save();
};
```

Reply:
Some people prefer more work for the developer and less for the compiler, but most don't.